src/components/dropdown/nz-dropdown.component.ts

```diff
@@ -112,49 +127,67 @@ export class NzDropDownComponent implements OnInit, OnDestroy, AfterViewInit {
     this._triggerWidth = this._nzOrigin.elementRef.nativeElement.getBoundingClientRect().width;
   }
 
-  _show() {
-    this._$mouseSubject.next(true);
+  _onVisibleChange = (visible: boolean) => {
+    if (visible) {
+      if (!this._triggerWidth) {
+        this._setTriggerWidth();
+      }
+    }
+    this.nzVisible = visible;
+    this._changeDetector.markForCheck();
   }
 
-  _hide() {
-    this._$mouseSubject.next(false);
+  _startSubscribe(observable$: Observable<boolean>) {
+    this._subscription = observable$
+      .subscribe(this._onVisibleChange)
   }
 
   ngOnInit() {
-    debounceTime.call(this._$mouseSubject, 300).subscribe((data: boolean) => {
-      this.nzVisible = data;
-      if (this.nzVisible) {
-        if (!this._triggerWidth) {
-          this._setTriggerWidth();
-        }
-      }
-      this.nzVisibleChange.emit(this.nzVisible);
-    });
     this._nzMenu.setDropDown(true);
   }
 
   ngOnDestroy() {
-    this._$mouseSubject.unsubscribe();
+    this._subscription.unsubscribe();
   }
 
-
   ngAfterViewInit() {
+    let mouse$: Observable<boolean>
     if (this.nzTrigger === 'hover') {
-      this._renderer.listen(this._nzOrigin.elementRef.nativeElement, 'mouseenter', () => this._show());
-      this._renderer.listen(this._nzOrigin.elementRef.nativeElement, 'mouseleave', () => this._hide());
+      mouse$ = Observable.create((observer: Observer<boolean>) => {
+        const disposeMouseEnter = this._renderer.listen(this._nzOrigin.elementRef.nativeElement, 'mouseenter', () => {
```

Comment:
should check whether will fire ngzone. although this component will not check

Reply:
Zone propagation for RxJS has been [fixed on Zone.js side](https://github.com/angular/zone.js/issues/830) already.